### ISSUE TITLE
Discussion Boxstarter script

### ISSUE THREAD

**Issue by vbjay** [gitextensions/gitextensions#5113]
I was looking at https://boxstarter.org/.  I have created this script so far.  Thought it would be a good to know and a way to script your dev setups.  

```powershell
try {
  cinst p4merge
  cinst git
  cinst gitextensions
 
    Write-ChocolateySuccess 'gitdev'
} catch {
  Write-ChocolateyFailure 'gitdev' $($_.Exception.Message)
  throw
}
```
One of my thoughts is that we could take over the chocolatey package.

**Comment by vbjay** [comment]
https://chocolatey.org/packages/gitextensions

**Comment by pmiossec** [comment]
* Change 'cinst' by 'choco install' which is now the recommended syntax
* I don't know how boxstarter works but surely that you should also use the '-y' option to remove user confirmation required.
* git is already a dependency of gitextensions package so you could remove the line... 
* perhaps a better message than just 'gitdev' 

I have never used boxstarter even if I was interested on it. It could be interesting...

**Comment by vbjay** [comment]
Right. Just getting started with it. Thanks. I really wanted to see if we
could get the chocalatey package.

**Comment by pmiossec** [comment]
While we talk about chocolate, I think that our release process should handle the generation and upload of the chocolatey package. What do you think?

I do it for another project that way user could update more easily (even if it's less important for gitextensions due to the auto update).

**Comment by RussKie** [comment]
> git is already a dependency of gitextensions package so you could remove the line...

No, we unbundled git and kdiff from both v2.51 and v3 installations in #4515

**Comment by pmiossec** [comment]
>No, we unbundled git and kdiff from both v2.51 and v3 installations in #4515

I would like to write something clearer and I just saw that I didn't :-(

Git is already a dependency of the gitextensions **chocolatey** package. That means that git will be installed by chocolatey before gitextensions even if git is bundled or not in the gitextensions package.

**Comment by vbjay** [comment]
Right. Once it's installed, our update can handle it but having it in
chocolatey would help new installs.  I think making it part of our release
process would help.

**Comment by vbjay** [comment]
> Change 'cinst' by 'choco install' which is now the recommended syntax

I'm going by https://github.com/chocolatey/boxstarter

**Comment by gep13** [comment]
>@vbjay said...
I'm going by https://github.com/chocolatey/boxstarter

I have an outstanding task to update everything in Boxstarter docs to recommend using `choco install` rathar than `cinst`.  To be clear, for now, `cinst` will work fine, but at some point, those aliases will likely go away, so better to start off on the right foot and use `choco install`.  Or, perhaps better `choco upgrade`, which will install the package if not already installed, and if it is installed, will check for a newer version.

As a side note, you shouldn't need to use `Write-ChocolateySuccess` or `Write-ChocolateyFailure` those cmdlets are now actually deprecated.

>@pmiossec said...
While we talk about chocolate, I think that our release process should handle the generation and upload of the chocolatey package. What do you think?

If you are interested in becoming a maintainer of this Chocolatey Package, then that is certainly something that we can help with.

**Comment by RussKie** [comment]
...if we can automate it - why not :)

**Comment by gep13** [comment]
@RussKie let us, the Chocolatey Team, know if you need any help getting this setup. We also like packages coming from the original author of an application.

**Comment by vbjay** [comment]
Thanks for the pointers.
